packages/kit/src/core/sync/write_client_manifest.js

```diff
@@ -43,25 +43,32 @@ export function write_client_manifest(manifest_data, output) {
 		})
 		.join(',\n\t');
 
+	const layouts_with_server_load = new Set();
+
 	const dictionary = `{
 		${manifest_data.routes
 			.map((route) => {
 				if (route.page) {
 					const errors = route.page.errors.slice(1).map((n) => n ?? '');
-					const layouts = route.page.layouts.slice(1).map((n) => {
-						if (n == undefined) {
-							return '';
-						}
-						return get_node_id(manifest_data.nodes, n);
-					});
+					const layouts = route.page.layouts.slice(1).map((n) => n ?? '');
 
 					while (layouts.at(-1) === '') layouts.pop();
 					while (errors.at(-1) === '') errors.pop();
 
-					const array = [get_node_id(manifest_data.nodes, route.page.leaf)];
+					// Encode whether or not the route uses server data
+					// using the ones' complement, to save space
+					const array = [`${route.leaf?.server ? '~' : ''}${route.page.leaf}`];
+					// Encode whether or not the layout uses server data.
+					// It's a different method compared to pages because layouts
+					// are reused across pages, so we safe space by doing it this way.
+					route.page.layouts.forEach((layout) => {
+						if (layout != undefined && manifest_data.nodes[layout].server) {
+							layouts_with_server_load.add(layout);
+						}
+					});
 
-					// only include non-root layout/error nodes if they exist
-					if (layouts.length > 0 || errors.length > 0) array.push(`[${layouts.join(',')}]`);
+					array.push(`[${layouts.join(',')}]`);
```

Comment:
Just realized, with the manifest adjustment the if that was there before can be reinstated